### ISSUE TITLE
Exposing rendering context to Nunjucks shortcodes fails in Eleventy 3.0.0-alpha.17

### ISSUE THREAD

**Issue by groenroos** [11ty/eleventy#3368]
### Operating system

macOS Sonoma 14.4.1

### Eleventy

3.0.0-alpha.17

### Describe the bug

Following the technique in [this comment](https://github.com/11ty/eleventy/issues/1596#issuecomment-1493049806), I've created a shortcode that loads a partial within a post, and injects it with the rendering context. Simplified example:

<details>
<summary><code>post.md</code></summary>

```markdown
---
title: Post
---

Text ...

{% promo "newsletter" %}

More text ...
```

</details>

<details>
<summary><code>.eleventy.js</code></summary>

```javascript
import addPromo from './lib/promo.js';

export default function(eleventyConfig) {
	/* Some collection */
	eleventyConfig.addCollection('stuff', function(collection) {
		return collection.getFilteredByTag('stuff');
	});

	/* Add helper */
	addPromo(eleventyConfig);

	return {
		markdownTemplateEngine: "njk",
	};
};
```

</details>

<details>
<summary><code>lib/promo.js</code></summary>

```javascript
import fs from 'fs';
import { EleventyRenderPlugin } from '@11ty/eleventy';

const RenderManager = EleventyRenderPlugin.RenderManager;

function addPromo(eleventyConfig) {
	const rm = new RenderManager();

	eleventyConfig.on('eleventy.config', cfg => {
		rm.templateConfig = cfg;
	});

	eleventyConfig.addAsyncShortcode('promo', async function (promoType) {
		const fn = await rm.compile(fs.readFileSync('./_includes/promo.njk').toString(), 'njk');
		return fn({ ...this.ctx.getAll(), promoType });
	});
}

export default addPromo;
```

</details>

<details>
<summary><code>_includes/promo.njk</code></summary>

```nunjucks
<h1>Sign up for our {{ promoType }}!</h1>
```

</details>

This works perfectly in Eleventy 2.0.0, but in Eleventy 3.0.0-alpha.17, it fails with an unhelpful error:

```
Problem writing Eleventy templates: (more in DEBUG output)
1. Having trouble rendering njk template ./post.md (via TemplateContentRenderError)
2. (./post.md)
  EleventyShortcodeError: Error with Nunjucks shortcode `promo` (via Template render error)
3. Error in your Eleventy config file '.eleventy.js'. (via Template render error)

Original error stack trace: EleventyConfigError: Error in your Eleventy config file '.eleventy.js'.
    at TemplateConfig.requireLocalConfigFile (/node_modules/@11ty/eleventy/src/TemplateConfig.js:376:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async TemplateConfig.mergeConfig (/node_modules/@11ty/eleventy/src/TemplateConfig.js:404:41)
    at async TemplateConfig.init (/node_modules/@11ty/eleventy/src/TemplateConfig.js:199:17)
    at async RenderManager.init (/node_modules/@11ty/eleventy/src/Plugins/RenderPlugin.js:394:3)
    at async RenderManager.compile (/node_modules/@11ty/eleventy/src/Plugins/RenderPlugin.js:424:3)
    at async Object.<anonymous> (/lib/promo.js:14:14)
```

Changing `...this.ctx.getAll()` to just `...this.ctx` seems to result in the object I want; but via a `try catch` I can surface this additional error that seems to happen as many times as the shortcode is used in the project:

```
EleventyConfigError: Error in your Eleventy config file '.eleventy.js'.
    at TemplateConfig.requireLocalConfigFile (/node_modules/@11ty/eleventy/src/TemplateConfig.js:376:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async TemplateConfig.mergeConfig (/node_modules/@11ty/eleventy/src/TemplateConfig.js:404:41)
    at async TemplateConfig.init (/node_modules/@11ty/eleventy/src/TemplateConfig.js:199:17)
    at async RenderManager.init (/node_modules/@11ty/eleventy/src/Plugins/RenderPlugin.js:394:3)
    at async RenderManager.compile (/node_modules/@11ty/eleventy/src/Plugins/RenderPlugin.js:424:3)
    at async Object.<anonymous> (/lib/promo.js:15:15) {
  originalError: UserConfigError: config.addCollection(stuff) already exists. Try a different name for your collection.
      at UserConfig.addCollection (/node_modules/@11ty/eleventy/src/UserConfig.js:1051:10)
      at default (/.eleventy.js:110:17)
      at TemplateConfig.requireLocalConfigFile (/node_modules/@11ty/eleventy/src/TemplateConfig.js:368:26)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async TemplateConfig.mergeConfig (/node_modules/@11ty/eleventy/src/TemplateConfig.js:404:41)
      at async TemplateConfig.init (/node_modules/@11ty/eleventy/src/TemplateConfig.js:199:17)
      at async RenderManager.init (/node_modules/@11ty/eleventy/src/Plugins/RenderPlugin.js:394:3)
      at async RenderManager.compile (/node_modules/@11ty/eleventy/src/Plugins/RenderPlugin.js:424:3)
      at async Object.<anonymous> (/lib/promo.js:15:15)
}
```

It appears to (somehow) try to re-register the first collection listed in `.eleventy.js`, but I'm not entirely sure how or why it's doing that.

### Reproduction steps

1. Create a project as per the above example files
2. Build it with Eleventy 3.0.0-alpha.17
3. Build will fail

### Expected behavior

Build should not fail; or at least a more useful error should be returned.

**Comment by zachleat** [comment]
Fix shipping with 3.0.0-alpha.18

**Comment by groenroos** [comment]
Can confirm that this is fixed in `3.0.0-alpha.18`, as long as `this.ctx.getAll()` is changed to `this.ctx`! Thank you! 🎉
